### PR TITLE
feat: Creates a custom attribute table

### LINKED ISSUE #2869: feat: Custom attribute table
- Add 2 tabs called `Conversation attributes` and `Contact attributes`
- Add a table under the tabs
- On clicking contact attributes, show the list of all the contact custom attributes in the table
- On clicking conversation attributes, show list all the conversation custom attributes in the table
- Supports pagination and filter
- Show edit/delete button 

### Investigation
- [ ] Figure out the max limit of custom attributes (Intercom limit: 250)

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

Creates a custom attribute table that includes the data and dummy buttons.

Fixes #2869 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

**Screen Recording**

https://user-images.githubusercontent.com/64252451/132158068-4f118e6c-5b0f-4868-81eb-5dce6b588a5f.mov

## How Has This Been Tested?

Please import `import attributes from './attributes/attributes.routes';` and add `...attributes.routes,` in `settings.routes.js` file.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/javascript/dashboard/api/attributes.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
+/* global axios */
 import ApiClient from './ApiClient';
 
 class AttributeAPI extends ApiClient {
   constructor() {
     super('custom_attribute_definitions', { accountScoped: true });
   }
+
+  getAttributeModel(modelId) {
```
Comment (muhsin-k) [inline app/javascript/dashboard/api/attributes.js:9]: Unused API method

File: app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+
+export default {
+  data() {
+    return {
+      selectedTabIndex: 0,
+      selectedAttribute: {},
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue:87]: Unused variable

File: app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+      const attributeModel = this.selectedTabIndex
+        ? 'contact_attribute'
+        : 'conversation_attribute';
+
+      return this.$store.getters['attributes/getAttributeModel'](
+        attributeModel
+      );
+    },
+    tabs() {
+      const tabs = [
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue:104]: No need for creating a variable. Just return the array

File: app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+          name: this.$t('ATTRIBUTES_MGMT.TABS.CONVERSATION'),
+        },
+        {
+          key: 1,
+          name: this.$t('ATTRIBUTES_MGMT.TABS.CONTACT'),
+        },
+      ];
+      return tabs;
+    },
+    selectedTabKey() {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue:116]: Unused computed property

File: app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  margin-top: var(--space-small);
+}
+
+.no-items-error-message {
+  margin-top: var(--space-larger);
+}
+
+.tabs {
+  padding-left: 0;
+  margin-right: 2.4rem;
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue:146]: CSS variable

File: app/javascript/dashboard/store/modules/attributes.js [hunk 6]
```diff
@@ -14,15 +14,18 @@ export const getters = {
   getUIFlags(_state) {
     return _state.uiFlags;
   },
-  getAttributes: _state => attributeType => {
+  getAttributes: _state => {
+    return _state.records;
+  },
+  getAttributeModel: _state => attributeModel => {
```
Comment (muhsin-k) [inline app/javascript/dashboard/store/modules/attributes.js:20]:
```suggestion
  getAttributesByModel: _state => attributeModel => {
```

**Review by muhsin-k — CHANGES_REQUESTED** [review]
@iamsivin Nice work 👍. Please see the inline comments

File: app/javascript/dashboard/api/attributes.js (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/api/attributes.js]: Now it is using, by direct calling `get()`, it is not showing contact attributes.

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**